Prox/Prox/PlaceDetails/PlaceDetailViewController.swift

```diff
@@ -97,52 +110,242 @@ class PlaceDetailViewController: UIViewController {
+            self.imageCarousel.alpha = 0
+            previousCardImageCarousel.alpha = 1
+            self.view.layoutIfNeeded()
+            }, completion: { finished in
+            if finished {
+                self.currentCardViewController.cardView.removeGestureRecognizer(self.panGestureRecognizer)
+                if let nextCardViewController = self.nextCardViewController {
+                    nextCardViewController.cardView.removeFromSuperview()
+                    nextCardViewController.cardView.isHidden = false
+                    nextCardViewController.removeFromParentViewController()
+                    self.unusedPlaceCardViewControllers.append(nextCardViewController)
+                }
+                self.imageCarousel.removeFromSuperview()
+                self.imageCarousel = previousCardImageCarousel
+                self.nextCardViewController = self.currentCardViewController
+                self.currentCardViewController = previousCardViewController
+                self.currentCardViewController.cardView.addGestureRecognizer(self.panGestureRecognizer)
+                self.previousCardViewController = newPreviousCardViewController
+            }
+        })
     }
+
+
+    fileprivate func insertNewPreviousViewController(toPreviousCard previousCardViewController: PlaceDetailsCardViewController) -> PlaceDetailsCardViewController? {
+        // deactivate and remove previous view controller trailing constraint
+        if let previousCardViewTrailingConstraint = previousCardViewTrailingConstraint {
+            NSLayoutConstraint.deactivate([previousCardViewTrailingConstraint])
+        }
+
+        guard let newNext = dataSource?.previousPlace(forPlace: previousCardViewController.place) else {
+            previousCardViewTrailingConstraint = nil
+            return nil
+        }
+
+        let newPreviousCardViewController = dequeuePlaceCardViewController(forPlace:newNext)
+        self.view.addSubview(newPreviousCardViewController.cardView)
+        self.addChildViewController(newPreviousCardViewController)
+        self.previousCardViewTrailingConstraint = newPreviousCardViewController.cardView.trailingAnchor.constraint(equalTo: previousCardViewController.cardView.leadingAnchor, constant: -cardViewSpacingConstant)
+        NSLayoutConstraint.activate([newPreviousCardViewController.cardView.topAnchor.constraint(equalTo: view.topAnchor, constant: cardViewTopAnchorConstant),
+                                     newPreviousCardViewController.cardView.widthAnchor.constraint(equalToConstant: cardViewWidthConstant),
+                                     self.previousCardViewTrailingConstraint!], translatesAutoresizingMaskIntoConstraints: false)
+
+        return newPreviousCardViewController
+    }
+
+    fileprivate func setupConstraints(forNewNextCard newNextCard: PlaceDetailsCardViewController, toNewCurrentCard newCurrentCard: PlaceDetailsCardViewController) {
```

Comment:
They were originally all one big function that ended up being far too complicated to follow. I'll rethink the naming here and make it consistent so it is easier to follow what is going on